File: php/features.php
<?php
declare(strict_types=1);

// Arbeitnow SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ArbeitnowFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ArbeitnowBaseFeature();
            case "test":
                return new ArbeitnowTestFeature();
            default:
                return new ArbeitnowBaseFeature();
        }
    }
}

File: py/features.py
# Arbeitnow SDK feature factory

from feature.base_feature import ArbeitnowBaseFeature
from feature.test_feature import ArbeitnowTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArbeitnowBaseFeature(),
        "test": lambda: ArbeitnowTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
